test/e2e/storage/volume_metrics.go

```diff
@@ -123,3 +175,22 @@ func getControllerStorageMetrics(ms metrics.ControllerManagerMetrics) map[string
 	}
 	return result
 }
+
+// Verifies the specified metrics are in `kubeletMetrics`
+func verifyVolumeStatMetric(metricKeyName string, namespace string, pvcName string, kubeletMetrics metrics.KubeletMetrics) {
+	found := false
+	if samples, ok := kubeletMetrics[metricKeyName]; ok {
+		for _, sample := range samples {
+			samplePVC, ok := sample.Metric["persistentvolumeclaim"]
+			Expect(ok).To(BeTrue(), "Error getting pvc for %s", metricKeyName)
+			sampleNS, ok := sample.Metric["namespace"]
+			Expect(ok).To(BeTrue(), "Error getting namespace for %s", metricKeyName)
+
+			if string(samplePVC) == pvcName && string(sampleNS) == namespace {
```

Comment:
Can we actually verify if emitted metrics match allocated PVC size?

Reply:
Would make sense. See my reply to @jeffvance 's comment below - I will follow up with more tests here in a separate PR in terms of verifying the metrics themselves.